src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -1027,17 +1489,181 @@ public class Derived2 : Derived
+                );
+            verify(comp);
+            comp = CreateCompilationWithCovariantReturns(source).VerifyDiagnostics(
+                );
+            verify(comp);
+            verify(CompilationReferenceView(comp));
+            verify(MetadataView(comp));
+
+            static void verify(CSharpCompilation comp)
+            {
+                VerifyOverride(comp, "Derived.M", "System.IComparable Derived.M()", "System.Object Base.M()");
+                VerifyOverride(comp, "Derived2.M", "System.String Derived2.M()", "System.IComparable Derived.M()");
+                VerifyAssignments(comp);
+            }
+        }
+
+        [Fact]
+        public void TestVBConsumption_01()
+        {
+            var source0 = @"
+public class Base
+{
+    public virtual object M() => null;
+    public virtual object P => null;
+    public virtual object this[int i] => null;
+}
+public abstract class Derived : Base
+{
+    public override string M() => null;
+    public override string P => null;
+    public override string this[int i] => null;
+}
+";
+            var csComp = CreateCompilationWithCovariantReturns(source0).VerifyDiagnostics(
+                );
+            csComp.VerifyDiagnostics();
+            var csRef = csComp.EmitToImageReference();
+
+            var vbSource = @"
+Public Class Derived2 : Inherits Derived
+    Public Overrides Function M() As String
+        Return Nothing
+    End Function
+    Public Overrides ReadOnly Property P As String
+        Get
+            Return Nothing
+        End Get
+    End Property
+    Public Overrides Default ReadOnly Property Item(i As Integer) As String
+        Get
+            Return Nothing
+        End Get
+    End Property
+    
+    Public Sub T(b as Base, d as Derived, d2 as Derived2)
+        Dim x1 As Object = b.M()
+        Dim x2 As Object = b.P
+        Dim x3 As Object = b(0)
+        Dim x4 As String = d.M()
+        Dim x5 As String = d.P
+        Dim x6 As String = d(0)
+        Dim x7 As String = d2.M()
+        Dim x8 As String = d2.P
+        Dim x9 As String = d2(0)
+    End Sub
+End Class
+";
+            var vbComp = CreateVisualBasicCompilation(code: vbSource, referencedAssemblies: csComp.References.Append(csRef));
+            vbComp.VerifyDiagnostics();
+            var vbTree = vbComp.SyntaxTrees[0];
+            var model = vbComp.GetSemanticModel(vbTree);
+            int count = 0;
+            foreach (var localDeclaration in vbTree.GetRoot().DescendantNodes().OfType<VisualBasic.Syntax.LocalDeclarationStatementSyntax>())
+            {
+                foreach (var declarator in localDeclaration.Declarators)
+                {
+                    count++;
+                    var initialValue = declarator.Initializer.Value;
+                    var typeInfo = model.GetTypeInfo(initialValue);
+                    Assert.True(typeInfo.Type.Equals(typeInfo.ConvertedType, TypeCompareKind.AllIgnoreOptions));
```

Comment:
>Equals [](start = 46, length = 6)

What API is called here?

Reply:
This is calling `Test.Utilities.Equals(this ITypeSymbol first, ITypeSymbol second, Microsoft.CodeAnalysis.TypeCompareKind typeCompareKind)`
However, the ignore options aren't needed so this is changed to `Assert.Equal(typeInfo.Type, typeInfo.ConvertedType)` in the following PR.


---
In reply to: [422144339](https://github.com/dotnet/roslyn/pull/43797#discussion_r422144339) [](ancestors = 422144339)